atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DbKeyValueServiceConfig.java

```diff
@@ -29,9 +42,53 @@ public DbSharedConfig shared() {
         return ImmutableDbSharedConfig.builder().build();
     }
 
+    public abstract ConnectionConfig connection();
+
     @Value.Derived
     public abstract Supplier<DbTableFactory> tableFactorySupplier();
 
-    public abstract SqlConnectionSupplier sqlConnectionSupplier();
+    @Value.Default
+    public Supplier<SqlConnectionSupplier> sqlConnectionSupplier() {
+        return () -> createConnectionSupplier();
```

Comment:
It really does look quite funky when you write it out.  Is there any way we can have the function name demonstrate the somewhat odd return type...